packages/expo/android/build.gradle

```diff
@@ -1,10 +1,41 @@
+import groovy.json.JsonSlurper
+
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 apply plugin: 'maven-publish'
 
 // Import autolinking script
 apply from: "../scripts/autolinking.gradle"
 
+static def versionToNumber(major, minor, patch) {
+  return patch * 100 + minor * 10000 + major * 1000000
+}
+
+def getRNVersion() {
+  def reactNativePackageJsonPath = [
+      "node",
+      "-e",
+      "console.log(require.resolve('react-native/package.json'));"
+  ]
+      .execute([], projectDir)
+      .text
+      .trim()
+
+  def packageJsonFile = new File(reactNativePackageJsonPath)
+  def packageJson = new JsonSlurper().parseText(packageJsonFile.text)
+  def version = safeExtGet("reactNativeVersion", packageJson.version)
```

Comment:
neat: i found @esamelson's approach to [get the version directly from nodejs](https://github.com/expo/expo/pull/16441/files#diff-4fa7841d1fd49df86f44c26076d9dc6752f3c183c2b659e1243b8b8c3d223f41R157-R159) is clever. maybe it's worth to take a look.